src/appleseed/renderer/modeling/project/projectfilereader.cpp

```diff
@@ -3065,6 +3065,9 @@ auto_release_ptr<Project> ProjectFileReader::read(
         }
 
         string unpacked_project_directory(project_filepath);
+
+        size_t ext_start = unpacked_project_directory.rfind(".");
```

Comment:
To be robust, you would need to check if the file has an extension. It's not guaranteed since you allow to open any zip file regardless of its extension.

I would suggest to use Boost with something like

    const string unpacked_project_directory =
        bf::path(project_filepath).stem().string() + ".unpacked";